apps/web/src/components/Twitter/TwitterFollowingModal.tsx

```diff
@@ -0,0 +1,213 @@
+          last: $twitterListLast
+          socialAccountType: Twitter
+        ) @connection(key: "TwitterFollowingModal__socialConnections") {
+          edges {
+            node {
+              __typename
+              ... on SocialConnection {
+                __typename
+                galleryUser {
+                  ... on GalleryUser {
+                    username
+                    bio
+                    ...FollowButtonUserFragment
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    followingRef
+  );
+
+  const { hideModal } = useModalActions();
+
+  const handleClose = useCallback(() => {
+    hideModal();
+  }, [hideModal]);
+
+  const twitterFollowing = useMemo(() => {
+    const users = [];
+
+    for (const edge of followingPagination?.data?.socialConnections?.edges ?? []) {
+      if (edge?.node?.__typename === 'SocialConnection' && edge?.node?.galleryUser) {
+        users.push(edge?.node.galleryUser);
+      }
+    }
+
+    return users;
+  }, [followingPagination]);
+
+  const [followAll] = usePromisifiedMutation<TwitterFollowingModalMutation>(graphql`
+    mutation TwitterFollowingModalMutation {
+      followAllSocialConnections(accountType: Twitter) {
+        __typename
+        ... on FollowAllSocialConnectionsPayload {
+          __typename
+          viewer {
+            ... on Viewer {
+              __typename
+              user {
+                ... on GalleryUser {
+                  __typename
+                  following {
+                    __typename
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  `);
+  const reportError = useReportError();
+  const { pushToast } = useToastActions();
+
+  const handleFollowAll = useCallback(async () => {
+    try {
+      await followAll({
+        variables: {},
+      });
+    } catch (error) {
+      if (error instanceof Error) {
+        reportError(error);
+        pushToast({
+          message: 'Unfortunately there was an error while following all users',
+        });
+      }
+    }
+  }, [followAll, pushToast, reportError]);
+
+  return (
+    <StyledOnboardingTwitterModal>
+      <StyledBodyTextContainer>
+        <TitleDiatypeL>
+          We've found {twitterFollowing.length}
+          {twitterFollowing.length === 1 ? ' person' : ' people'} you know from Twitter
+        </TitleDiatypeL>
+      </StyledBodyTextContainer>
+
+      <StyledFollowingContainer gap={16}>
+        {twitterFollowing.map((user) => (
+          <HStack key={user.username} align="center" justify="space-between">
+            <VStack>
+              <StyledLink
+                href={{
+                  // @ts-expect-error TODO: fix noImplicitAny error here
+                  pathname: '/[username]',
+                  query: {
+                    username: user.username as string,
+                  },
+                }}
+              >
```

Comment:
@tbezman not sure why but im getting ts `Query` error on this. Not sure what Im doing wrong here 🤦🏻‍♂️

Reply:
Hmmm, there's some weird type issue with styled components and the Next.js Link where it's props are somehow getting Tanstack's Query involved. Not sure what's going on there but I fixed it by manually telling the styled component what type the props should be.